packages/botonic-react/src/webchat-app.jsx

```diff
@@ -111,25 +111,32 @@ export class WebchatApp {
     return this.hubtypeService.postMessage(user, input)
   }
 
-  async resendUnsentInputs() {
-    return this.hubtypeService.resendUnsentInputs()
+  async onConnectionRegained() {
+    return this.hubtypeService.onConnectionRegained()
   }
 
   onStateChange({ session: { user }, messagesJSON }) {
-    if (!this.hubtypeService && user) {
-      const lastMessage = messagesJSON[messagesJSON.length - 1]
-      this.hubtypeService = new HubtypeService({
-        appId: this.appId,
-        user,
-        lastMessageId: lastMessage && lastMessage.id,
-        lastMessageUpdateDate: this.getLastMessageUpdate(),
-        onEvent: event => this.onServiceEvent(event),
-        unsentInputs: () =>
-          this.webchatRef.current
-            .getMessages()
-            .filter(msg => msg.ack === 0 && msg.unsentInput),
-        server: this.server,
-      })
+    const lastMessage = messagesJSON[messagesJSON.length - 1]
+    const lastMessageId = lastMessage && lastMessage.id
+    const lastMessageUpdateDate = this.getLastMessageUpdate()
+    if (!this.hubtypeService) {
```

Comment:
nit: this might seem very picky, but it improves the readability. I'd start with the simple case and add the other condition altogether next:

```
if (this.hubtypeService) {
   ...
} else if (!this.hubtypeService && user) {
   ...
}
```

Reply:
cool! 😎